src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -624,7 +613,7 @@ private void CheckDisposed()
                     }
                     finally
                     {
-                        connectionState.Connection.Transport.Input.AdvanceTo(consumed, examined);
+                        connectionState.Connection.Transport.Input.AdvanceTo(buffer.Start);
```

Comment:
From past experience explaining this to @BrennanConroy, put a comment in about how `buffer` was moved because it's passed by `ref`